packages/api/lambdas/sf-event-sqs-to-db-records/write-execution.js

```diff
@@ -1,69 +1,157 @@
-const log = require('@cumulus/common/log');
+const isNil = require('lodash/isNil');
+
 const {
   tableNames,
 } = require('@cumulus/db');
 const {
   getMessageExecutionArn,
+  getExecutionUrlFromArn,
   getMessageCumulusVersion,
+  getMessageExecutionOriginalPayload,
+  getMessageExecutionFinalPayload,
 } = require('@cumulus/message/Executions');
 const {
   getMetaStatus,
+  getMessageWorkflowTasks,
+  getMessageWorkflowName,
+  getMessageWorkflowStartTime,
+  getMessageWorkflowStopTime,
 } = require('@cumulus/message/workflows');
 
+const { parseException } = require('../../lib/utils');
 const Execution = require('../../models/executions');
 
-const {
-  isPostRDSDeploymentExecution,
-  hasNoAsyncOpOrExists,
-  hasNoParentExecutionOrExists,
-} = require('./utils');
+const shouldWriteExecutionToPostgres = ({
+  messageCollectionNameVersion,
+  collectionCumulusId,
+  messageAsyncOperationId,
+  asyncOperationCumulusId,
+  messageParentExecutionArn,
+  parentExecutionCumulusId,
+}) => {
+  const noMessageCollectionOrExists = isNil(messageCollectionNameVersion)
```

Comment:
```
noMessageCollectionOrExists
noMessageAsyncOperationOrExists
noMessageParentExecutionOrExists
```
It wasn't initially clear to me why these were named this way until you explained it. I understood the check for a null `messageCollectionNameVersion` and so on, but not the not null check. I think, for clarity, adding `InPostgres` to the variable names would make that more explicit.

```suggestion
  const noMessageCollectionOrExistsInPostgres = isNil(messageCollectionNameVersion)
    || !isNil(collectionCumulusId);
```

Reply:
Yeah, good feedback and suggestions. Made those changes here: 870aee35857b8f43b284a5cb57950e1c2895492d